### PR TITLE
UCP/RNDV: Fallback to PUT_ZCOPY or AM when memtype pipeline rndv failed

### PR DESCRIPTION
**Pull request by bureddy**

### CODE REVIEW

File: src/ucp/rndv/rndv.c [hunk 1]
```diff
@@ -1720,6 +1720,12 @@ static ucs_status_t ucp_rndv_send_start_put_pipeline(ucp_request_t *sreq,
         if (!(md_attr->cap.reg_mem_types & UCS_BIT(UCS_MEMORY_TYPE_HOST))) {
             return UCS_ERR_UNSUPPORTED;
         }
+
+        /* check if mem type endpoint is exists */
+        if (!UCP_MEM_IS_HOST(sreq->send.mem_type) &&
+            worker->mem_type_ep[sreq->send.mem_type] == NULL) {
```
Comment (dmitrygx) [inline src/ucp/rndv/rndv.c:1726]: wrap the 2nd condition by `()`

File: src/ucp/rndv/rndv.c [hunk 2]
```diff
@@ -1859,7 +1865,7 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_rndv_rtr_handler,
         if (is_pipeline_rndv) {
             status = ucp_rndv_send_start_put_pipeline(sreq, rndv_rtr_hdr);
             if (status != UCS_ERR_UNSUPPORTED) {
-                return status;
+                ucs_debug("pipeline_rndv is failed. fallback to PUT_ZCOPY or AM");
```
Comment (dmitrygx) [inline src/ucp/rndv/rndv.c:1868]: why need it? e.g. it can be `UCS_ERR_NO_MEMORY` - I guess we should fail operation if faced an unrecoverable error

**Comment by yosefe** [comment]
@bureddy  can you pls add "why" section?

File: src/ucp/rndv/rndv.c (hunk 2 above)
Comment (bureddy) [reply src/ucp/rndv/rndv.c:1868]: you are right.

**Comment by bureddy** [comment]
replace with https://github.com/openucx/ucx/pull/6454

**Pull request closed without merging**